### DISCUSSION TITLE
How to get the min / max value to rescale using a mosaicJSON for dem COGs

### DISCUSSION THREAD

**Discussion by tdurand** [developmentseed/titiler#701] [Q&A] [answered]
Hello, 

I am trying to use titiler with the elevation dataset Copernicus DEM : https://registry.opendata.aws/copernicus-dem/

I have produced a Mosaic JSON using `cogeo-mosaic` for the entire COG dataset (16 MB) : https://filedn.com/lkrqWbAQYllSVUK4ip6g3m0/mosaic.json  .  (using the [list of all the COGs here](https://copernicus-dem-30m.s3.amazonaws.com/tileList.txt))

Then I use titiler to get tiles from it , for example: [https://titiler.xyz/mosaicjson/tiles/12/1535/2322.png?url=https://filedn.com/lkrqWbAQYllSVUK4ip6g3m0/mosaic.json](https://titiler.xyz/mosaicjson/tiles/12/1535/2322.png?url=https://filedn.com/lkrqWbAQYllSVUK4ip6g3m0/mosaic.json)

But I get a grey tile

![2322](https://github.com/developmentseed/titiler/assets/533590/ccb7c2d4-8ee6-48c9-b1c1-a39fcb6d0be9)

If I use some rescale `rescale=0,2000` parameter to map the values to fit in the png format it works:

[https://titiler.xyz/mosaicjson/tiles/12/1535/2322.png?url=https://filedn.com/lkrqWbAQYllSVUK4ip6g3m0/mosaic.json&rescale=0,2000](https://titiler.xyz/mosaicjson/tiles/12/1535/2322.png?url=https://filedn.com/lkrqWbAQYllSVUK4ip6g3m0/mosaic.json&rescale=0,2000)

![2322](https://github.com/developmentseed/titiler/assets/533590/c4749d81-e74d-4ec3-be2a-d933e57dc58a)

But my issue is that I do not know what is the min, max to specify , is it the min max of the whole dataset ? or the min max of the bunch of COG files used by the mosaic for this tile ? 

For example for a single COG I can get the min/max info like this:  [https://titiler.xyz/cog/statistics?url=https://copernicus-dem-30m.s3.amazonaws.com/Copernicus_DSM_COG_10_N08_00_W068_00_DEM/Copernicus_DSM_COG_10_N08_00_W068_00_DEM.tif](https://titiler.xyz/cog/statistics?url=https://copernicus-dem-30m.s3.amazonaws.com/Copernicus_DSM_COG_10_N08_00_W068_00_DEM/Copernicus_DSM_COG_10_N08_00_W068_00_DEM.tif)

![image](https://github.com/developmentseed/titiler/assets/533590/321af512-9933-4d0c-9bba-3cf81feb18d3)

My question would be, what is the best way to set the rescale= parameter in case of a mosaic ? 

Thanks a lot

Thibault

**Comment by vincentsarago** [comment, marked as answer]
When using the rescaling parameter, this should be the min/max values for the whole dataset.

https://titiler.xyz/mosaicjson/map?url=https://filedn.com/lkrqWbAQYllSVUK4ip6g3m0/mosaic.json&rescale=0,6000&colormap_name=terrain 

☝️ tiles will start loading at zoom 8 

Note: we've added Copernicus DEM to [rio-tiler-pds](https://github.com/cogeotiff/rio-tiler-pds) which in theory remove the need of creating MosaicJSON for this dataset. 

```python
from rio_tiler_pds.copernicus.aws.dem import Dem30Reader 
from titiler.core.factory import TilerFactory

# We need to overwrite the default `url` path parameter
def path_dep() -> str:
    return "copernicus_dem"


from fastapi import FastAPI

app = FastAPI()

endpoints = TileFactory(
    reader=Dem30Reader,
    path_dependency=path_dep
)
app.include_router(endpoints.router)
```

**Reply by tdurand** [reply]
An up to date answer: https://github.com/cogeotiff/rio-tiler-pds/discussions/84#discussioncomment-13661742

**Comment by tdurand** [comment]
> When using the rescaling parameter, this should be the min/max values for the whole dataset.

Thanks ! Yes in the meantime I figured out that, I have set to 0,8900 I should be fine 😅

> Note: we've added Copernicus DEM to [rio-tiler-pds](https://github.com/cogeotiff/rio-tiler-pds) which in theory remove the need of creating MosaicJSON for this dataset.

Ahh good to know, I will have a look. anyways it was a good exercise to generate a mosaic json myself, will be useful for future needs.

Thanks a lot !
